diff --git a/daphne/daphne-1.0-src/vldp2/vldp/vldp_internal.c b/daphne/daphne-1.0-src/vldp2/vldp/vldp_internal.c
index 1b73d9d..5883964 100644
--- a/daphne/daphne-1.0-src/vldp2/vldp/vldp_internal.c
+++ b/daphne/daphne-1.0-src/vldp2/vldp/vldp_internal.c
@@ -29,7 +29,10 @@
 
 #include <stdint.h>
 
+#define __USE_POSIX
 #include <stdio.h>
+#undef __USE_POSIX
+
 #include <stdlib.h>	// for malloc/free
 #include <string.h>
 #include <sys/types.h>
diff --git a/daphne/src/audio/SDL_audiocvt.c b/daphne/src/audio/SDL_audiocvt.c
index 5f714c5..6c27b6f 100644
--- a/daphne/src/audio/SDL_audiocvt.c
+++ b/daphne/src/audio/SDL_audiocvt.c
@@ -19,6 +19,7 @@
   3. This notice may not be removed or altered from any source distribution.
 */
 #include <stdint.h>
+#include <math.h>
 #include "../SDL_internal.h"
 
 /* Functions for audio drivers to perform runtime conversion of audio format */
diff --git a/daphne/src/thread/pthread/SDL_systhread.c b/daphne/src/thread/pthread/SDL_systhread.c
index 5d97d62..f4e581e 100644
--- a/daphne/src/thread/pthread/SDL_systhread.c
+++ b/daphne/src/thread/pthread/SDL_systhread.c
@@ -28,12 +28,17 @@
 #endif
 
 #include <signal.h>
+#include <stdint.h>
 
 #ifdef __LINUX__
 #include <sys/time.h>
 #include <sys/resource.h>
+
+#define __USE_MISC
 #include <sys/syscall.h>
 #include <unistd.h>
+#undef __USE_MISC
+
 #endif /* __LINUX__ */
 
 #if defined(__LINUX__) || defined(__APPLE__)
@@ -98,7 +103,7 @@ SDL_SYS_CreateThread(SDL_Thread *thread, void *args)
     }
 
     /* Create the thread and go! */
-    if (pthread_create(&thread->handle, &type, RunThread, args) != 0) {
+    if (pthread_create((pthread_t * restrict)&thread->handle, &type, RunThread, args) != 0) {
 		// 2017.02.07 - RJS ADD - Logging.
         return SDL_SetError("Not enough resources to create thread");
     }
