diff --git a/custom/dependencies/libzlib/gzguts.h b/custom/dependencies/libzlib/gzguts.h
index d87659d..8963947 100644
--- a/custom/dependencies/libzlib/gzguts.h
+++ b/custom/dependencies/libzlib/gzguts.h
@@ -19,6 +19,7 @@
 #endif
 
 #include <stdio.h>
+#include <unistd.h>
 #include "zlib.h"
 #ifdef STDC
 #  include <string.h>
diff --git a/libretro/libretro.c b/libretro/libretro.c
index 3099025..7810293 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -719,7 +719,7 @@ void retro_init(void)
         initializing = true;
 
         retro_thread = co_active();
-        game_thread = co_create(65536 * sizeof(void*) * 16, EmuThreadFunction);
+        game_thread = co_create(65536 * sizeof(void*) * 16, (void(*)(void))EmuThreadFunction);
     }
 
     m64p_error ret = CoreStartup(FRONTEND_API_VERSION, ".", ".", NULL, n64DebugCallback, 0, n64StateCallback);
diff --git a/mupen64plus-core/subprojects/minizip/zip.c b/mupen64plus-core/subprojects/minizip/zip.c
index d54fe1c..c39b3f5 100644
--- a/mupen64plus-core/subprojects/minizip/zip.c
+++ b/mupen64plus-core/subprojects/minizip/zip.c
@@ -1246,7 +1246,7 @@ extern int ZEXPORT zipOpenNewFileInZip4_64 (zipFile file, const char* filename,
         unsigned char bufHead[RAND_HEAD_LEN];
         unsigned int sizeHead;
         zi->ci.encrypt = 1;
-        zi->ci.pcrc_32_tab = get_crc_table();
+        zi->ci.pcrc_32_tab = (const long unsigned int *)get_crc_table();
         /*init_keys(password,zi->ci.keys,zi->ci.pcrc_32_tab);*/
 
         sizeHead=crypthead(password,bufHead,RAND_HEAD_LEN,zi->ci.keys,zi->ci.pcrc_32_tab,crcForCrypting);
