diff --git a/Makefile b/Makefile
index 09eb547..8109905 100644
--- a/Makefile
+++ b/Makefile
@@ -175,6 +175,10 @@ else ifneq (,$(findstring rpi,$(platform)))
          CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a72
          ARM_CPUFLAGS = -mfpu=neon-fp-armv8
       endif
+   else ifneq (,$(findstring rpi5,$(platform)))
+      CPUFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76
+      GLES3= 1
+      GL_LIB := -lGLESv3
    else ifneq (,$(findstring rpi,$(platform)))
       CPUFLAGS += -mcpu=arm1176jzf-s
       ARM_CPUFLAGS = -mfpu=vfp
