diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index b153565..00cebda 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -1302,7 +1302,7 @@ static void frontend_unix_set_screen_brightness(int value)
 
    /* Device tree should have 'label = "backlight";' if control is desirable */
    filestream_read_file("/sys/class/backlight/backlight/max_brightness",
-                        &buffer, NULL);
+                        (void **)&buffer, NULL);
    if (buffer)
    {
       sscanf(buffer, "%u", &max_brightness);
diff --git a/configuration.c b/configuration.c
index 9e4705a..428f4e2 100644
--- a/configuration.c
+++ b/configuration.c
@@ -74,6 +74,10 @@
 #include "ui/drivers/cocoa/apple_platform.h"
 #endif
 
+#ifdef HAVE_LAKKA
+#include <time.h>
+#endif
+
 enum video_driver_enum
 {
    VIDEO_GL                 = 0,
diff --git a/network/drivers_wifi/connmanctl.c b/network/drivers_wifi/connmanctl.c
index 95ba029..13a6d3f 100644
--- a/network/drivers_wifi/connmanctl.c
+++ b/network/drivers_wifi/connmanctl.c
@@ -20,6 +20,7 @@
 #include <string/stdstring.h>
 #include <retro_miscellaneous.h>
 #include <time.h>
+#include <unistd.h>
 
 #include "../wifi_driver.h"
 
